modules/widgets/internet-defense-league.php

```diff
@@ -0,0 +1,154 @@
+			$this->variant = $this->defaults['variant'];
+		}
+		?>
+		<script type="text/javascript">
+			window._idl = {};
+			_idl.campaign = "<?php echo esc_js( $this->campaign ); ?>";
+			_idl.variant = "<?php echo esc_js( $this->variant ); ?>";
+			(function() {
+				var idl = document.createElement('script');
+				idl.type = 'text/javascript';
+				idl.async = true;
+				idl.src = ('https:' == document.location.protocol ? 'https://' : 'http://') + 'members.internetdefenseleague.org/include/?url=' + (_idl.url || '') + '&campaign=' + (_idl.campaign || '') + '&variant=' + (_idl.variant || 'banner');
+				document.getElementsByTagName('body')[0].appendChild(idl);
+			})();
+		</script>
+		<?php
+	}
+
+	public function form( $instance ) {
+		$instance = wp_parse_args( $instance, $this->defaults );
+
+		// Hide first two form fields if no current campaigns.
+		if ( false === $this->no_current ) {
+			echo '<p><label>';
+			echo __( 'Which Internet Defense League campaign do you want to participate in?', 'jetpack' ) . '<br />';
+			$this->select( 'campaign', $this->campaigns, $instance['campaign'] );
+			echo '</label></p>';
+
+			echo '<p><label>';
+			echo __( 'How do you want to promote the campaign?', 'jetpack' ) . '<br />';
+			$this->select( 'variant', $this->variants, $instance['variant'] );
+			echo '</label></p>';
+		}
+
+		echo '<p><label>';
+		echo __( 'Which badge would you like to display?', 'jetpack' ) . '<br />';
+		$this->select( 'badge', $this->badges, $instance['badge'] );
+		echo '</label></p>';
+
+		/* translators: %s is a name of an internet campaign called the "Internet Defense League" */
+		echo '<p>' . sprintf( _x( 'Learn more about the %s', 'the Internet Defense League', 'jetpack' ), '<a href="https://www.internetdefenseleague.org/">Internet Defense League</a>' ) . '</p>';
+	}
+
+	public function select( $field_name, $options, $default = null ) {
+		echo '<select class="widefat" name="' . $this->get_field_name( $field_name ) . '">';
+		foreach ( $options as $option_slug => $option_name ) {
+			echo '<option value="' . esc_attr( $option_slug ) . '"' . selected( $option_slug, $default, false ) . '>' . esc_html( $option_name ) . '</option>';
+		}
+		echo '</select>';
+	}
+
+	public function update( $new_instance, $old_instance ) {
+		$instance = array();
+
+		$instance['campaign'] = ( isset( $new_instance['campaign'] ) && isset( $this->campaigns[ $new_instance['campaign'] ] ) ) ? $new_instance['campaign'] : $this->defaults['campaign'];
+		$instance['variant']  = ( isset( $new_instance['variant'] )  && isset( $this->variants[  $new_instance['variant']  ] ) ) ? $new_instance['variant']  : $this->defaults['variant'];
+		$instance['badge']    = ( isset( $new_instance['badge'] )    && isset( $this->badges[    $new_instance['badge'] ] ) )    ? $new_instance['badge']    : $this->defaults['badge'];
+
+		return $instance;
+	}
+}
+
+function jetpack_internet_defense_league_init() {
+	if ( Jetpack::is_active() ) {
```

Comment:
This widget doesn't require anything from wpcom so it doesn't need a connection. This condition can be removed.